File: src/main/java/com/vinyl/model/Salesman.java
package com.vinyl.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.sql.Date;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
public class Salesman {
    private Integer tabNum;
    private String salesmanName;
    private String passportNum;
    private String addressCity;
    private String addressStr;
    private String addressHome;
    private Integer addressApt;
    private String salesmanPhoneNum;
    private Date worksFrom;
    private Date worksTo;
    private Integer salary;
    private String salesmanLogin;

}

File: src/main/resources/frontend/src/schemas/definitions/salesman.js
export default {
  props: {
    tabNum: {
      type: 'int',
      readonly: true
    },
    salesmanName: 'string',
    passportNum: 'string',
    addressCity: 'string',
    addressStr: 'string',
    addressHome: 'string',
    addressApt: {
      type: 'int',
      isNullable: true,
      typeConstraint: {
        from: 1
      }
    },
    salesmanPhoneNum: {
      type: 'string',
      typeConstraint: /^\+?\d+$/
    },
    worksFrom: {
      type: 'date',
      typeConstraint: {
        to: []
      }
    },
    worksTo: {
      type: 'date',
      isNullable: true,
      typeConstraint: {
        to: []
      }
    },
    salary: {
      type: 'int',
      typeConstraint: {
        from: 0
      }
    },
    checks: {
      type: 'check',
      isSchema: true,
      filterOnly: true,
      joins: ['cheq']
    }
  },
  key: 'tabNum',
  nodelete: true,
  display: (item) => item.salesmanName
}